hivemind/p2p/servicer.py

```diff
@@ -0,0 +1,90 @@
+import asyncio
+import dataclasses
+import importlib
+from functools import partial
+from typing import Any, Callable, Optional, Union
+
+from hivemind.p2p.p2p_daemon import P2P, P2PContext
+from hivemind.p2p.p2p_daemon_bindings.datastructures import PeerID
+
+
+@dataclasses.dataclass
+class RPCHandler:
+    method_name: str
+    handle_name: str
+    request_type: type
+    response_type: type
+
+
+class StubBase:
+    def __init__(self, p2p: P2P, peer: PeerID):
+        self.p2p = p2p
+        self.peer = peer
```

Comment:
The servicer derives stub classes for particular services (e.g. DHT, averager, etc.) from `StubBase`, adding the necessary `rpc_*` methods. The `.p2p` and `.peer` fields are somewhat private and used only inside this methods. Thus, it is not semantically a dataclass.

I will add a docstring clarifying that.

Reply:
I guess besides the docstring, it's at least worth converting these fields to private ones

Reply:
Fixed (initially, I was afraid that it will trigger linter warnings since methods using these fields are defined outside of the class, in the `Servicer._make_rpc_caller` method).